### PR TITLE
React Native Firebase Storage for Non-Expo Workflow

### PR DESCRIPTION
**Pull request by zolomohan**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by zolomohan** [comment]
This fixes #1154

**Comment by hectorkambow** [comment]
👋 Thank you for submitting this PR - we will review it ASAP - all PRs go into our queue will be looked over before being approved to be published.

**Review by WanjaMIKE — CHANGES_REQUESTED** [review]
Hey @zolomohan Nice article. Quite Informative. Could you please make the following minor changes

File: articles/react-native-firebase-storage/index.md [hunk 1]
```diff
@@ -0,0 +1,605 @@
...
+images:
+
+  - url: /engineering-education/react-native-firebase-storage/hero.jpg
+    alt: React Native Non-Expo Firebase Storage Hero Image
+---
+In this tutorial, we will be building a Non-Expo React Native application to upload images and videos to Firebase's cloud storage.
+<!--more-->
+
+### Firebase
+Firebase is a platform developed by Google for creating mobile and web applications. It was originally an independent company founded in 2011. In 2014, Google acquired the platform and it is now their flagship offering for app development.
```
Comment (WanjaMIKE) [inline articles/react-native-firebase-storage/index.md:21]: Provide a brief explanation of Firebase's cloud storage service here

File: articles/react-native-firebase-storage/index.md [hunk 2]
```diff
@@ -0,0 +1,605 @@
...
+You can take a look at the final code in this [GitHub Repository](https://github.com/zolomohan/react-native-firebase-storage-non-expo).
+
+### Development environment
+> **IMPORTANT** - We will not be using Expo in our project.
+
+You can follow [this documentation](https://reactnative.dev/docs/environment-setup) to set up the environment and create a new React app.
+
+Make sure you're following the React Native CLI Quickstart, not the Expo CLI Quickstart.
+
+![Env Setup](env_setup.png)
```
Comment (WanjaMIKE) [inline articles/react-native-firebase-storage/index.md:50]: Provide full image link

File: articles/react-native-firebase-storage/index.md [hunk 3]
```diff
@@ -0,0 +1,605 @@
...
+
+Now, click on the Android icon to add an android app to the Firebase project.
+
+![register_app](/engineering-education/react-native-firebase-storage/register_app.png)
+
+You will need the package name of the application to register the application. You can find the package name in the `AndroidManifest.xml` which is located in `android/app/src/main/`.
+
+![Package Name](/engineering-education/react-native-firebase-storage/package_name.png)
+
+Once you enter the package name and proceed to the next step, you can download the `google-services.json` file. You should place this file in the `android/app` directory.
```
Comment (WanjaMIKE) [inline articles/react-native-firebase-storage/index.md:103]: Provide an explanation of the importance of `google-services.json` file

File: articles/react-native-firebase-storage/index.md [hunk 4]
```diff
@@ -0,0 +1,605 @@
...
+
+On the next step, it will ask you for the storage bucket location.
+
+![Bucket Location](/engineering-education/react-native-firebase-storage/bucket_location.png)
+
+Once this is done, you'll see this screen. You can upload files and delete files manually from this screen.
+
+![Storage Setup Done](/engineering-education/react-native-firebase-storage/storage_setup_done.png)
+
+Now, Let's edit the cloud storage rules and set it to open. Switch to the `Rules` tab.
```
Comment (WanjaMIKE) [inline articles/react-native-firebase-storage/index.md:155]: put open in this ``open``

File: articles/react-native-firebase-storage/index.md [hunk 5]
```diff
@@ -0,0 +1,605 @@
...
+
+> The minimum target SDK for the React Native Image Picker is 21. If your project targets an SDK below 21, bump up the minSDK target in `android/build.gradle`.
+
+After the package is installed, import the `launchCamera` and `launchImageLibrary` functions from the package.
+
+```JSX
+import { launchCamera, launchImageLibrary } from 'react-native-image-picker';
+```
+
+Both functions accept 2 arguments. The first argument is `options` for the camera or the gallery, and the second argument is a callback function. This callback function is called when the user picks a media or cancels the operation.
```
Comment (WanjaMIKE) [inline articles/react-native-firebase-storage/index.md:258]: Could you talk more on what the function returns to determine the user's action or choice.

Comment (zolomohan) [reply articles/react-native-firebase-storage/index.md:260]: The value that the function returns is not important. I think those functions return nothing. That's why it's accepting callback functions. 

Discussing the return value of those functions provide no value to the topic.

**Comment by zolomohan** [comment]
@WanjaMIKE I've made those minor changes.

**Review by WanjaMIKE — APPROVED** [review]

**Comment by zolomohan** [comment]
@hectorkambow Just a heads up. This article could trigger plagiarism alerts from my previous articles since I reused some common sections.

**Comment by hectorkambow** [comment]
plagiarism check done  👁️

**Pull request merged**